parametric/apps/golang/otel_methods.go

```diff
@@ -0,0 +1,175 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	ot_api "go.opentelemetry.io/otel/trace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+func (s *apmClientServer) OtelStartSpan(ctx context.Context, args *OtelStartSpanArgs) (*OtelStartSpanReturn, error) {
+	fmt.Println("started_StartOtelSpan")
+
+	var pCtx context.Context = context.Background()
+	if pid := args.GetParentId(); pid != "" {
+		parent, ok := s.otelSpans[pid]
+		if ok {
+			pCtx = tracer.ContextWithSpan(context.Background(), parent.(ddtrace.Span))
+		}
+	}
+	var otelOpts = []ot_api.SpanStartOption{
+		ot_api.WithSpanKind(ot_api.ValidateSpanKind(ot_api.SpanKind(args.GetSpanKind()))),
+	}
+	if args.GetNewRoot() {
+		otelOpts = append(otelOpts, ot_api.WithNewRoot())
+	}
+	if t := args.GetTimestamp(); t != 0 {
+		tm := time.Unix(t, 0)
+		otelOpts = append(otelOpts, ot_api.WithTimestamp(tm))
+	}
+
+	// if attrs := args.GetAttributes(); attrs != nil {
```

Comment:
Current api allows passing in the attributes at the start, it's fin to continue with it